src/formatDate.js

```diff
@@ -50,7 +50,38 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitted = date.split(fromFormat[3]);
+  const year = toFormat.includes('YY') ? 'YY' : 'YYYY';
+  const formattedDate = {};
+  const result = [];
+
+  for (let i = 0; fromFormat.length - 1 > i; i++) {
+    for (let k = 0; toFormat.length - 1 > k; k++) {
+      if (toFormat[k].includes(fromFormat[i][0])) {
+        formattedDate[toFormat[k]] = splitted[i];
+      }
+    }
+  }
+
+  if (formattedDate.YYYY && formattedDate[year].length === 2) {
+    formattedDate[year] = formattedDate[year] < 30
+      ? 20 + formattedDate[year]
+      : 19 + formattedDate[year];
```

Comment:
move formattedDate[year] into const variable and use it instead of formattedDate[year], because there are a lot of duplicates